src/client/java/daos-java/src/main/native/daos_jni_common.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * (C) Copyright 2018-2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#include "daos_jni_common.h"
+
+/**
+ * This function is called when JVM load native library through
+ * System.loadLibrary or System.load methods.
+ *
+ * \param[in]	vm		denote a Java VM
+ * \param[in]	reserved	reserved for future
+ *
+ * \return	JNI_VERSION expected by JVM on success return.
+ * 		JNI_ERR or non-zeor rc code of daos_init() for any error. In
+ * 		this case, JVM throws JNI error.
+ */
+jint
+JNI_OnLoad(JavaVM *vm, void *reserved)
+{
+	JNIEnv *env;
+
+	if ((*vm)->GetEnv(vm, (void **)&env, JNI_VERSION) != JNI_OK) {
+		return JNI_ERR;
+	}
+
+	jclass local_class = (*env)->FindClass(env,
+			"io/daos/DaosIOException");
+
+	daos_io_exception_class = (*env)->NewGlobalRef(env, local_class);
+	jmethodID m1 = (*env)->GetMethodID(env, daos_io_exception_class,
+			"<init>",
+			"(Ljava/lang/String;)V");
+
+	new_exception_msg = (*env)->NewGlobalRef(env, m1);
+	if(new_exception_msg == NULL){
+		printf("failed to get constructor msg\n");
+		return JNI_ERR;
+	}
+	jmethodID m2 = (*env)->GetMethodID(env, daos_io_exception_class,
+			"<init>",
+			"(Ljava/lang/Throwable;)V");
+
+	new_exception_cause = (*env)->NewGlobalRef(env, m2);
+	if(new_exception_cause == NULL){
```

Comment:
(style)  space required before the open brace '{'
(style)  space required before the open parenthesis '('